back-end/dao/projectsDAO.js

```diff
@@ -1,112 +1,158 @@
-    }catch(e){
-        console.error(`Unable to issue insertOne command, ${e}`);
+    /**
+     * Create a project object in MongoDB (see AddProject.js in /front-end for more info).
+     * @param req
+     * @param res
+     * @param next
+     * @returns {Promise<void>}
+     */
+    static async createProject(req, res, next) {
+        try {
+            await projects.insertOne(req);
+        } catch (e) {
+            console.error(`Unable to issue insertOne command, ${e}`);
+        }
     }
-  }
 
-  static async deleteProject(projectName) {
-    try {
-      const deleteResponse = await projects.deleteOne({
-        project_name: projectName,
-      });
-      return deleteResponse;
-    } catch (e) {
-      console.error(`Unable to delete project: ${e}`);
-      return { error: e };
+    /**
+     * Delete the project object with the name projectName from the database.
+     * @param projectName
+     * @returns {Promise<{error}|*>}
+     */
+    static async deleteProject(projectName) {
+        try {
+            const deleteResponse = await projects.deleteOne({
+                project_name: projectName,
+            });
+            return deleteResponse;
+        } catch (e) {
+            console.error(`Unable to delete project: ${e}`);
+            return {error: e};
+        }
     }
-  }
 
-  static async updateProject(projectName, transcript) {
-    try {
-      const updateResponse = await projects.updateOne(
-        { project_name: projectName },
-        { $addToSet: { transcripts: transcript } }
-      );
 
-      return updateResponse;
-    } catch (e) {
-      console.error(`Unable to update project: ${e}`);
-      return { error: e };
+    /**
+     *
+     * @param projectName
+     * @param transcript
+     * @returns {Promise<{error}|*>}
+     */
+    static async updateProject(projectName, transcript) {
+        try {
+            const updateResponse = await projects.updateOne(
+                {project_name: projectName},
+                {$addToSet: {transcripts: transcript}}
+            );
+
+            return updateResponse;
+        } catch (e) {
+            console.error(`Unable to update project: ${e}`);
+            return {error: e};
+        }
     }
-  }
 
-  /**
-   * Get a project object with a particular id from MongoDB.
-   * @param id
-   */
-  static async getProjectByID(id) {
-    try {
-      const pipeline = [
-        {
-          $match: {
-            _id: new ObjectId(id),
-          },
-        },
-        {
-          $lookup: {
-            from: "text_transcripts",
-            let: {
-              id: "$_id",
-            },
-            pipeline: [
-              {
-                $match: {
-                  $expr: {
-                    $eq: ["$project_id", "$$id"],
-                  },
+    /**
+     * Get a project object with a particular id from MongoDB.
+     * Also, add parsed and trimmed transcripts corresponding to that project to the project object.
```

Comment:
Nice work on the documentation!

I was wondering if we could split this method into two? I.e., one method that gets a project object with a particular ID from MongoDB, and another that adds parsed and trimmed transcripts corresponding to a specific project. 

That way, we could have the option of getting a project with a particular ID without adding transcripts to it. I wonder if this is why the Upload Transcripts use case is **duplicating transcripts that were already added to the project** (as discussed on Discord)?